lib/util/git/onboarding-branch-cache.ts

```diff
@@ -0,0 +1,42 @@
+import type { ExtractResult } from '../../workers/repository/process/extract-update';
+import { getCache } from '../cache/repository';
+import type { OnboardingBranchCache } from '../cache/repository/types';
+
+export function getCachedOnboardingBranch(
+  branchSha: string,
+  baseBranchSha: string,
+  baseBranchName: string
+): OnboardingBranchCache | null {
+  const cache = getCache();
+  const { onboardingBranch } = cache;
+  if (!onboardingBranch || onboardingBranch?.sha === null) {
+    return null;
+  }
+
+  onboardingBranch.parentSha ??= cache.scan?.[baseBranchName].sha;
+  if (
+    onboardingBranch?.sha !== branchSha ||
+    onboardingBranch.parentSha !== baseBranchSha
+  ) {
+    return null;
+  }
+  return onboardingBranch;
+}
+
+export function setOnboardingBranchCache(
+  branchSha: string,
+  baseBranchSha: string,
+  isOnboarded: boolean,
+  extractedDependencies?: ExtractResult
+): void {
+  const cache = getCache();
+  const { onboardingBranch = {} as OnboardingBranchCache } = cache;
```

Comment:
are you sure we need the cast? if yes:

```suggestion
  const onboardingBranch = cache.onboardingBranch ?? {};
```

Reply:
onboardingBranch is optional property so without type-cast then TS will throw error.